### DISCUSSION TITLE
BluetoothA2DPSink.h: No such file or directory

### DISCUSSION THREAD

**Discussion by MuAlRi** [pschatzmann/ESP32-A2DP#632] [Q&A] [answered]
I'm currently making a simple project that is Bluetooth speaker using ESP32 DEVKIT V1,
Components:
ESP32 DEVKIT V1
PAM8610
Speaker 12Watt

I'm currently having a problem having this code in the library, as simple as uploading the zip file it does not show that it was included. I ask some AI saying that the structure of the file is not correct as i need the src file that shows at the examples at Arduino Ide. I also ask guidance to my professor, he says try to change the file to Ino type files and individually put them on to their folders, it didn't shows on the examples at Arduino Ide but i moved it from src to examples in this file it shows at the examples in Arduino Ide

The file that shows in the examples at Arduino at the examples at this file. using my professor method the src file that I individually put them on to their folders, changing them to Ino type files and moving the to examples on this file actually shows at the examples in Arduino Ide but when i verify/compile it, it's still shows the same Eror

If anyone have a resolve to this, Please kindly respond
Thankyou for your time.

**Comment by pschatzmann** [comment, marked as answer]
There is no way around __installing this library correctly__!
After you did this (and maybe restarted Arduino) you will see the examples in Arduino under ESP32-A2DP!

You can also double check the installation in your libraries folder: the content must match with what you see on Github.

The Readme gives a suggestion how to install

**Reply by MuAlRi** [reply]
<img width="960" alt="ss" src="https://github.com/user-attachments/assets/2bb1c522-c123-437f-b757-28b9300014f3">

What i did is only uploading it to my library and check it through examples, but when i compile it's still shows the same Eror

**Reply by pschatzmann** [reply]
What error ?

**Reply by MuAlRi** [reply]
BluetoothA2DPSink.h: No such file or directory

**Reply by pschatzmann** [reply]
Did you double check your libraries directory ?

**Reply by MuAlRi** [reply]
I did it shows in the library, the problem is after i compile it's still shows the same eror BluetoothA2DPSink.h: No such file or directory. i even did a re-install on ArduinoIde, and did a redo

**Reply by pschatzmann** [reply]
Show me the exact error log and the expanded content of the libraries folder

**Reply by MuAlRi** [reply]
<img width="960" alt="image" src="https://github.com/user-attachments/assets/43d319e5-0298-432e-9392-f76935babb24">

![image](https://github.com/user-attachments/assets/7cd8c19e-3676-4ca1-bb9d-616a4c1ced14)

![image](https://github.com/user-attachments/assets/5ed0671f-10fb-43c8-a151-58756164916a)

Here's some screenchot

**Reply by pschatzmann** [reply]
ps. I see that you forgot to install the audio-tools project

**Reply by pschatzmann** [reply]
and what is the content of this ?  Expand the content until you see what's inside the src directory

**Reply by MuAlRi** [reply]
What i did is a simple Upload the Zip file to the include library

![image](https://github.com/user-attachments/assets/f10a3ca7-81aa-47ab-b43c-3cb4aedfc82b)

and i'm quite new to this, so is there a file that i forgot to upload, iwas just following the steps

**Reply by pschatzmann** [reply]
and what is in the src directory ?

**Reply by MuAlRi** [reply]
![image](https://github.com/user-attachments/assets/0262a146-1d1f-4a68-b09e-1b1ebcbba489)

**Reply by pschatzmann** [reply]
Is it normal that you don't see the file extensions ?
Why is there an Error in your file path ? Error > Nathan...

Apart from this, this is what I would have expected to see

**Reply by MuAlRi** [reply]
Oh it's just a sync eror from one drive, does it make a different?

![image](https://github.com/user-attachments/assets/7a37b51f-2b21-40a8-9466-38297856e338)

**Reply by pschatzmann** [reply]
Just google Arduino and one-drive !
I would move the arduino directory to your disk,,,

**Reply by MuAlRi** [reply]
I'm not quite understand, what do you mean by not seeing the file extensions

**Reply by pschatzmann** [reply]
I guess you are using window and I dont. In Linux I see the following

![image](https://github.com/user-attachments/assets/ddbcd5c1-cb13-4b2a-892f-7d701bdbfe23)

**Reply by MuAlRi** [reply]
It is on my disk, if you see the file path. The Eror just saying the deleted files from my computer and wanted to deleted it on one drive too or not?. am i correct?

**Reply by MuAlRi** [reply]
Ah i see

**Reply by pschatzmann** [reply]
Just google Arduino and one-drive !
I would move the arduino directory to your disk,,,

**Reply by MuAlRi** [reply]
does it make the diffrent in ArduinoIde if im using windows?

**Reply by pschatzmann** [reply]
Windows should work: but putting your arduino directory on one-drive is asking for trouble!

**Reply by MuAlRi** [reply]
i see you what you mean, let me try to change the sketchbook location
im following this forum
https://forum.arduino.cc/t/arduino-and-onedrive/1150807/3
